### ISSUE TITLE
OpenShift startup fails (on first start)

### ISSUE THREAD

**Issue by hferentschik** [projectatomic/adb-utils#26]
```
$ sudo systemctl enable openshift
$ sudo systemctl start openshift
Job for openshift.service failed because the control process exited with error code. See "systemctl status openshift.service" and "journalctl -xe" for details.

$ sudo journalctl --unit=openshift
Feb 15 09:21:36 localhost.localdomain systemd[1]: openshift.service: main process     exited, code=exited, status=143/n/a
Feb 15 09:21:36 localhost.localdomain systemd[1]: Failed to start Docker     Application Container for OpenShift.
Feb 15 09:21:36 localhost.localdomain systemd[1]: Unit openshift.service entered     failed state.
Feb 15 09:21:36 localhost.localdomain systemd[1]: openshift.service failed.
Feb 15 09:22:06 localhost.localdomain systemd[1]: openshift.service holdoff time     over, scheduling restart.
Feb 15 09:22:06 localhost.localdomain systemd[1]: Starting Docker Application     Container for OpenShift...
Feb 15 09:22:06 localhost.localdomain docker[13560]: openshift
Feb 15 09:22:07 localhost.localdomain docker[13565]: openshift
Feb 15 09:22:07 localhost.localdomain docker[13571]: Error response from daemon:     Conflict: Tag latest is already set to image     62208f15133798af253a651386ae2753da9130fcc68f1863cbd735c547356173, if you want to     replace it, please use -f opt
Feb 15 09:22:08 localhost.localdomain sh[13667]: [INFO] Starting OpenShift server    
```

**Comment by vrutkovs** [comment]
AFAIK this is related to [status check](https://github.com/projectatomic/adb-utils/blob/master/services/openshift/scripts/openshift_provision#L13): if container didn't appear in 30 secs the curl will return exit code '1' and we'll be attempting to show logs for non-existant container.

We should put curl call in a loop (instead of 'sleep 30' in service) and show logs only if container exists

**Comment by hferentschik** [comment]
Actually in openshift_vagrant we did something like this:

```
########################################################################
wait_for_openshift() {
  # Give OpenShift time to start
  for i in {1..6}
  do
    curl -ksSf https://${1}:8443/api > /dev/null 2>&1
    if [ $? -ne 0 ]; then
      sleep 5
    else
      break
    fi
  done

  # Final check whether OpenShift is running
  curl -ksSf https://{$1}:8443/api > /dev/null 2>&1
  if [ $? -ne 0 ]; then
    >&2 echo "[ERROR] OpenShift failed to start:"
    docker logs ose
    exit 1
  fi
}
```

That said, one would assume that the check whether openshift is running is happening in _/opt/adb/openshift/openshift_. There should really be no need anymore for this check once _openshift_provision_ is called.

BTW, I am skeptical about the use of 127.0.0.1. The internal OpenShift IP is 10.0.2.15.

**Comment by vrutkovs** [comment]
Right, did a similar thing in PR #36. 127.0.0.1 would work too - as long as its in `corsAllowedOrigins` in `master-config.yaml`

**Comment by praveenkumar** [comment]
I tested this PR and it was failing after bit of debug I found out patch which @vrutkovs created should be part of `openshift_provision` instead `openshift` script. 

Reason for failure: As soon as https://github.com/projectatomic/adb-utils/blob/master/services/openshift/scripts/openshift#L157 kicks in systemd start ExecPost section and `wait_for_openshift_api` will be uncalled and failed with logs. 

I am going to put PR which make this patch to part of `openshift_provision`.
